frontend/src/views/HomeView.vue

```diff
@@ -1 +1,69 @@
-<template><h1>Test</h1></template>
+<script lang="ts" setup>
+import { onMounted, ref } from 'vue'
+import { ElButton, ElDialog, ElInput } from 'element-plus'
+import { useRolesStore } from '@/stores/roles/index'
+import { ErrorMessage } from '@/services/messages/messages'
+import { SuccessMessage } from '@/services/messages/messages'
+import { HttpStatusCode } from '@/services/http'
+
+const dialogFormVisible = ref(false)
+const rolesStore = useRolesStore()
+const newRoleName = ref('')
+
+onMounted(async () => {
+  await rolesStore.fetchRoles()
+})
+
+const onClickCreateRole = () => {
+  dialogFormVisible.value = true
+}
+
+const onClickCancel = () => {
+  dialogFormVisible.value = false
+}
+
+const createRole = async () => {
+  if (newRoleName.value == '') {
+    ErrorMessage('You need to fill in this field')
+    dialogFormVisible.value = true
```

Comment:
You could rename the functions to show/hide the dialog, and use it here also (since it's the same logic)